android/src/com/mapswithme/maps/settings/StorageUtils.java

```diff
@@ -66,16 +60,12 @@ static String getWritableDirRoot()
 
   static long getFreeBytesAtPath(String path)
   {
-    long size = 0;
-    try
-    {
-      size = new File(path).getFreeSpace();
-    } catch (RuntimeException e)
-    {
-      e.printStackTrace();
-    }
+    StatFs statFs = new StatFs(path);
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2)
+      return statFs.getFreeBlocksLong() * statFs.getBlockSizeLong();
```

Comment:
@deathbaba Documentation states that File.getFreeSpace() is "likely to be an optimistic over-estimate and should not be taken as a guarantee your application can actually write this many bytes".
@ygorshenin You're right. getAvailableBlocksLong() is more appropriate here.